x/fungible/keeper/foreign_coins.go

```diff
@@ -81,20 +81,18 @@ func (k Keeper) GetAllForeignCoins(ctx sdk.Context) (list []types.ForeignCoins)
 	return
 }
 
-// GetAllForeignERC20CoinMap returns all foreign ERC20 coins in a map of chainID -> asset -> coin
-func (k Keeper) GetAllForeignERC20CoinMap(ctx sdk.Context) map[int64]map[string]types.ForeignCoins {
+// GetAllForeignCoinMap returns all foreign ERC20 coins in a map of chainID -> asset -> coin
```

Comment:
Can we add a note this must not be used outside of queries?
Using it in keeper logic might create non-determinism condition because of the map

Reply:
https://github.com/zeta-chain/node/pull/2060/commits/fa24f32191e070f8729095afe027e573ac1a3d51